metagraph/src/cli/query.cpp

```diff
@@ -264,65 +267,86 @@ Json::Value SeqSearchResult::to_json(bool verbose_coords,
-                              utils::join_strings(std::get<LabelVec>(result),
-                                                  delimiter));
-    } else if (std::holds_alternative<LabelCountVec>(result)) {
+        output += fmt::format("\t{}", utils::join_strings(*v, delimiter));
+
+    } else if (const auto *v = std::get_if<LabelCountVec>(&result_)) {
         // Labels with count data
-        for (const auto &[label, count] : std::get<LabelCountVec>(result)) {
+        for (const auto &[label, count] : *v) {
             output += fmt::format("\t<{}>:{}", label, count);
         }
-    } else if (std::holds_alternative<LabelSigVec>(result)) {
+    } else if (const auto *v = std::get_if<LabelSigVec>(&result_)) {
         // Count signatures
-        for (const auto &[label, kmer_presence_mask] : std::get<LabelSigVec>(result)) {
+        for (const auto &[label, kmer_presence_mask] : *v) {
             output += fmt::format("\t<{}>:{}:{}:{}", label,
                                   sdsl::util::cnt_one_bits(kmer_presence_mask),
                                   sdsl::util::to_string(kmer_presence_mask),
                                   anno_graph.score_kmer_presence_mask(kmer_presence_mask));
         }
-    } else if (std::holds_alternative<LabelQuantileVec>(result)) {
-        // Count quantiles
-        for (const auto &[label, quantiles] : std::get<LabelQuantileVec>(result)) {
-            output += fmt::format("\t<{}>:{}", label, fmt::join(quantiles, ":"));
+    } else if (const auto *v = std::get_if<LabelAbundanceVec>(&result_)) {
+        // k-mer counts (or quantiles)
+        for (const auto &[label, counts] : *v) {
+            output += "\t<" + label + ">";
+            if (verbose_output) {
+                output += fmt::format(":{}", fmt::join(counts, ":"));
+            } else {
+                std::pair<size_t, size_t> last(0, counts.at(0));
+                for (size_t i = 1; i <= counts.size(); ++i) {
+                    if (i < counts.size() && counts[i] == last.second)
+                        continue; // extend run
+
+                    // end of run
+                    if (last.second) {
+                        if (i == last.first + 1) {
+                            output += fmt::format(":{}={}", last.first, last.second);
+                        } else {
+                            output += fmt::format(":{}-{}={}", last.first, i - 1, last.second);
+                        }
+                    }
+
+                    // start new run
+                    if (i < counts.size()) {
+                        last.first = i;
+                        last.second = counts[i];
+                    }
+                }
+            }
         }
-    } else if (std::holds_alternative<LabelCoordVec>(result)) {
+    } else {
         // Kmer coordinates
-        for (const auto &[label, tuples] : std::get<LabelCoordVec>(result)) {
+        for (const auto &[label, tuples] : std::get<LabelCoordVec>(result_)) {
```

Comment:
again, if we add a new annotation type without remembering to change this, this'll segfault

Reply:
no, it will throw, so you'll notice